use watchers and data instead of computed props to memoize results

indirectlylit opened the pull request:
### Summary

Computed props are re-evaluated down the entire chain, whether or not it's necessary. This can be particularly expensive when the prop drives styles or DOM updates.

| before | after |
|--|--|
| ![before](https://user-images.githubusercontent.com/2367265/46573090-1f0f3800-c95e-11e8-9c61-538f9ec9b87b.gif) | ![after](https://user-images.githubusercontent.com/2367265/46573091-21719200-c95e-11e8-8ac1-0c23797ded06.gif) |

### Reviewer guidance

make sure responsive behaviors are still working, especially grids

### References

* fixes #4075
* re: https://github.com/vuejs/vue/issues/8540

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

indirectlylit committed:
Revert "set vue to a beta version for testing"

This reverts commit 355acf387266f6b0095cd7461e496485268079ba.

indirectlylit committed:
updating responsive window mixin to only use watchers

indirectlylit committed:
re-enable test

indirectlylit committed:
remove unnecessary complexity

jonboiser commented:
The SetupWizardIndex test is failing because of 'DefaultLanguageForm'. You can stub DefaultLanguageForm in the tests, or figure out what the problem is in DefaultLanguageForm.

jonboiser committed:
Fix SetupWizardIndex test

jonboiser committed:
Remove stub

jonboiser committed:
Remove un-used autofocus

Having this autosave attr breaks FacilityPermissionsForm-related tests for some reason

indirectlylit commented:
thanks for addressing those issues @jonboiser 

was there anything else that I should be doing to get this PR in?

indirectlylit commented:
thanks!
